### PR TITLE
mgr/zabbix: Implement health checks

### PR DESCRIPTION
**Pull request by wido**
This PR adds health checks for the Zabbix Mgr module to notify the admin if something goes wrong with sending data to Zabbix.

### CODE REVIEW

File: src/pybind/mgr/zabbix/module.py [hunk 1]
```diff
@@ -231,9 +240,17 @@ def send(self):
...
 
             zabbix.send(identifier, data)
+            self.set_health_checks(dict())
             return True
         except Exception as exc:
             self.log.error('Exception when sending: %s', exc)
+            self.set_health_checks({
+                'MGR_ZABBIX_SEND_FAILED': {
+                    'severity': 'warning',
+                    'summary': 'Failed to send data to Zabbix due to a Exception',
```
Comment (jcsp) [inline src/pybind/mgr/zabbix/module.py]: It would leave off the "due to a Exception here" -- the message conveys all the useful information without it.

**Comment by wido** [comment]
Good and valid point. Fixed that one

**Comment by wido** [comment]
For the QA:

<pre>
root@alpha:~# rm /usr/bin/zabbix_sender 
root@alpha:~# ceph -s
  cluster:
    id:     6e4daf97-c6a1-47c9-9606-6dd92c527cd2
    health: HEALTH_WARN
            Failed to send data to Zabbix due to a Exception
 
  services:
    mon: 3 daemons, quorum alpha,bravo,charlie
    mgr: alpha(active)
    osd: 3 osds: 3 up, 3 in
 
  data:
    pools:   1 pools, 64 pgs
    objects: 201 objects, 201 MB
    usage:   3763 MB used, 26653 MB / 30416 MB avail
    pgs:     64 active+clean
 
root@alpha:~# ceph health detail
HEALTH_WARN Failed to send data to Zabbix due to a Exception
MGR_ZABBIX_SEND_FAILED Failed to send data to Zabbix due to a Exception
    [Errno 2] No such file or directory
root@alpha:~#
</pre>

I removed the *zabbix_sender* binary and after a minute the health went to WARN because it failed to execute.

**Comment by tchaikov** [comment]
@wido ahh, just noticed your comment !

@wido and @jcsp how shall we address the failure of http://pulpito.ceph.com/kchai-2018-02-01_09:17:08-rados-wip-kefu-testing-2018-02-01-1258-distro-basic-mira/2137803/ ? shall we whitelist this warning or prepare a (dummy) zabbix_sender in the module_selftest?

**Comment by jcsp** [comment]
@tchaikov I think we can blacklist it for the selftest task.

**Comment by jcsp** [comment]
@wido please could you update qa/suites/rados/mgr/tasks/module_selftest.yaml in this PR, to add the new health message to the list of whitelisted log messages

**Comment by wido** [comment]
Done @jcsp :) This should whitelist all Zabbix related things

File: qa/suites/rados/mgr/tasks/module_selftest.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ tasks:
         - objects misplaced
         - Synthetic exception in serve
         - influxdb python module not found
+        - \(MGR_ZABBIX_\)
```
Comment (tchaikov) [inline qa/suites/rados/mgr/tasks/module_selftest.yaml]: this does not work, see http://pulpito.ceph.com/kchai-2018-02-05_11:48:09-rados-wip-kefu-testing-2018-02-05-1650-distro-basic-mira/2155066/ . 

what we want to whitelist is cluster log like
```
[WRN] Health check failed: Failed to send data to Zabbix (MGR_ZABBIX_SEND_FAILED)
```

so `(MGR_ZABBIX_)` does not match.  but `(MGR_ZABBIX_` does. see
```
echo '[WRN] Health check failed: Failed to send data to Zabbix (MGR_ZABBIX_SEND_FAILED)' | egrep '\(MGR_ZABBIX'
[WRN] Health check failed: Failed to send data to Zabbix (MGR_ZABBIX_SEND_FAILED)
```

probably you could use `\(MGR_ZABBIX_` or `\(MGR_ZABBIX_SEND_FAILED\)` instead.

**Comment by wido** [comment]
Done @tchaikov ! Should be fixed. Could you run the tests again?

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-02-11_04:16:47-rados-wip-kefu-testing-2018-02-11-0959-distro-basic-mira/

 failures are tracked by

- https://tracker.ceph.com/issues/21218
- https://tracker.ceph.com/issues/21312
- https://tracker.ceph.com/issues/22979

**Pull request merged**